### DISCUSSION TITLE
How to refresh the component - Same Facebook comment is shown in all posts

### DISCUSSION THREAD

**Discussion by dibasdauliya** [vercel/next.js#30965] [Help]
Hi, I have integrated [Facebook Comment](https://developers.facebook.com/docs/plugins/comments/) Plugin and [Facebook page plugin](https://developers.facebook.com/docs/plugins/page-plugin/) into [my website](https://www.readersclubnepal.com/categories/story/in-war-of-ego-loser-always-wins)

Problem 1: When the website [opens at first](https://www.readersclubnepal.com) it renders the Facebook page plugin but while navigating across pages it doesn't render.
I guess prefetching things of `<Link>` is doing this. 

Problem 2: On the home page, there is no FB comment plugin. When we click some articles then there is an FB comment plugin but it doesn't render, we should refresh the page and then It renders the comment plugin.
Also, if I comment in one post then the same comment is shown in each and every post :( 

This is Fb page component:
```jsx
import React from 'react'
import Script from 'next/script'

export const LikeFbPage = () => {
  return (
    <>
      <Script
        async
        // defer
        crossOrigin='anonymous'
        src='https://connect.facebook.net/en_US/sdk.js#xfbml=1&version=v12.0&appId=xxxx&autoLogAppEvents=1'
        nonce='WbcKWdmK'
      ></Script>

      <div className='htt'>
        <div
          className='fb-page'
          data-href='https://www.facebook.com/xxx'
          data-tabs=''
          data-width=''
          data-height=''
          data-small-header='false'
          data-adapt-container-width='true'
          data-hide-cover='false'
          data-show-facepile='true'
        >
          <blockquote
            cite='https://www.facebook.com/xxx'
            className='fb-xfbml-parse-ignore'
          >
            <a href='https://www.facebook.com/xxx'>
              xxx
            </a>
          </blockquote>
        </div>
      </div>
    </>
  )
}

```

And, I have placed `<div id='fb-root'></div>` in `_document.js`

I'm calling `<LikeFbPage/>` on different pages but it only shows at first and while navigating across other pages it disappears and shows only while refreshing

**Comment by Silverium** [comment]
https://developers.facebook.com/docs/reference/javascript/FB.XFBML.parse/
```js
  useEffect(() => {
    FB.XFBML.parse();
  }, []);
```
